src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -598,6 +598,16 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			this._userNameInputBox.enable();
 			this._passwordInputBox.enable();
 			this._rememberPasswordCheckBox.enabled = true;
+
+			const recentConnections = this._connectionManagementService.getRecentConnections();
```

Comment:
Is this actually where we want the logic to be? It'll affect everything opening the connection dialog - which may be problematic for example if it was opened with a different password pre-populated for some reason which would then be overwritten by this. 

Could you just move this logic to where the azure pane is making the connection to instead do the lookup and population of the password field there which is then passed into here?